crates/rmcp/src/transport/child_process.rs

```diff
@@ -41,13 +42,35 @@ pub struct TokioChildProcess {
 }
 
 pub struct ChildWithCleanup {
-    inner: Box<dyn TokioChildWrapper>,
+    inner: Option<Box<dyn TokioChildWrapper>>,
 }
 
 impl Drop for ChildWithCleanup {
     fn drop(&mut self) {
-        if let Err(e) = self.inner.start_kill() {
-            tracing::warn!("Failed to kill child process: {e}");
+        // Close child more graceful
+        if let Some(mut child) = self.inner.take() {
+            // Spawn a background task to clean up
```

Comment:
So the child will eventually be cleaned up. For certain applications however it would be nice if the caller / API user would have a way to have a more controlled way to clean up / shutdown, something like `child.shutdown().await` that also returns the error instead of just logging it.